packages/admin-panel/src/surveyResponse/FileQuestionField.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import generateId from 'uuid/v1';
+import { Button, Dialog, DialogFooter, DialogHeader, TextField } from '@tupaia/ui-components';
+import { getUniqueFileNameParts } from '@tupaia/utils';
+import EditIcon from '@material-ui/icons/Edit';
+import DeleteIcon from '@material-ui/icons/Delete';
+import ExportIcon from '@material-ui/icons/GetApp';
+import { FileUploadField } from '../widgets/InputField/FileUploadField';
+import { download } from '../VizBuilderApp/api';
+import { IconButton, ModalContentProvider } from '../widgets';
+
+const Container = styled.div`
+  padding-bottom: 1.2rem;
+`;
+
+const TextFieldWithActions = styled.div`
+  display: flex;
+  gap: 15px;
+`;
+
+const Actions = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 3px;
+`;
+
+const SmallIconButton = styled(IconButton)`
+  scale: 0.8;
+`;
+
+const generateUniqueFileName = fileName => `${generateId()}_${fileName}`;
```

Comment:
Would really love to find a way for us to share these kind of utils between app/server side. Let's chat separately 👍

Reply:
Discussed, going with doing this in a followup card